tests/moduletool/test_release.py

```diff
@@ -0,0 +1,376 @@
+def test_bump_dev_version(
+    tmpdir, modules_dir: str, modules_v2_dir: str, monkeypatch, v1_module: bool, increment: str, changelog_file_exists: bool
+) -> None:
+    """
+    Ensure that the normal scenario for the `inmanta module release --dev` command works as expected.
+    """
+    module_name = "mod"
+    path_module = os.path.join(tmpdir, module_name)
+    if v1_module:
+        v1_module_from_template(
+            source_dir=os.path.join(modules_dir, "minimalv1module"),
+            dest_dir=path_module,
+            new_version=Version("1.1.2.dev0"),
+            new_name=module_name,
+        )
+    else:
+        module_from_template(
+            source_dir=os.path.join(modules_v2_dir, "minimalv2module"),
+            dest_dir=path_module,
+            new_version=Version("1.1.2.dev0"),
+            new_name=module_name,
+        )
+
+    if changelog_file_exists:
+        path_changelog_file = os.path.join(path_module, const.MODULE_CHANGELOG_FILE)
+        with open(path_changelog_file, "w", encoding="utf-8") as fh:
+            fh.write(
+                """
+V1.1.2
+- Test
+
+V1.1.1
+- Mentioning the other version 1.1.2, 1.2.0, 2.0.0
+
+V1.1.0
+- Release
+
+V1.0.0
+- Release
+            """.strip()
+            )
+
+    gitprovider.git_init(repo=path_module)
+    gitprovider.commit(repo=path_module, message="Initial commit", add=["*"], commit_all=True)
+    for tag in ["1.0.0", "1.1.0", "1.1.1", "3.0.0"]:
+        gitprovider.tag(repo=path_module, tag=tag)
+
+    def assert_module_state(expected_version: str, expected_commit_message: str) -> None:
+        """
+        Verify that the module has the state as expected after running the `inmanta module release command`.
+
+        :param expected_version: The expected fully qualified version (version number + version tag).
+        :param expected_commit_message: The commit message expected for the latest commit.
+        """
+        assert Module.from_path(path_module).version == Version(expected_version)
+        path_changelog_file = os.path.join(path_module, const.MODULE_CHANGELOG_FILE)
+        if changelog_file_exists:
+            with open(path_changelog_file, "r", encoding="utf-8") as fh:
+                # Ensure that only the first occurrence of the version number in the
+                # changelog file got replaced.
+                stable_version_number = expected_version.rsplit(".", maxsplit=1)[0]
+                assert len(re.findall(re.escape(stable_version_number), fh.read())) == 2
+        else:
+            assert not os.path.exists(path_changelog_file)
+        assert get_commit_message_x_commits_ago(path=path_module, nb_previous_commit=0).strip() == expected_commit_message
+
+    monkeypatch.chdir(path_module)
+    module_tool = ModuleTool()
+    module_tool.release(dev=True, patch=True, message="Commit patch increment")
+    assert_module_state("1.1.2.dev0", expected_commit_message="Initial commit")
```

Comment:
I can't figure out why the latest commit isn't "Commit patch increment"

Reply:
The previous stable release of the module is version  `1.1.1` and the current version of the module is `1.1.2.dev0`. So they are already a patch increment separated from each other. When executing `inmanta module release --dev --patch -m "Commit patch increment"`, no commit is required to bump the version. As such, there is no commit with the message `Commit patch increment`. Should I log a warning saying that no version bump was required?

Reply:
I see it now :+1: